src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -246,4 +253,63 @@ public Migration migrated(HttpServletRequest request) {
 		}
 		return rv;
 	}
+	
+	/************* Box integration *****************/
+	/**
+	 * get json string og box folders
+	 * @return
+	 */
+	private String DEVELOPER_TOKEN = "";
+	private static final int MAX_DEPTH = 1;
+    
+	@RequestMapping("/box/folders")
+	public String getBoxFolders() {
+		
+		String rv = "";
+
+		// TODO: get the DEVELOPER_TOKEN from configuratio for now, will do token auto-generation in the future
+		DEVELOPER_TOKEN = env.getProperty("box_api_token");
+		
+		BoxAPIConnection api = new BoxAPIConnection(DEVELOPER_TOKEN);
+		BoxUser.Info userInfo = BoxUser.getCurrentUser(api).getInfo();
+		log.info("Box user log in success: user name = " + userInfo.getName() + " user login = " + userInfo.getLogin());
+		
+		BoxFolder rootFolder = BoxFolder.getRootFolder(api);
+		List<BoxFolder> folders = listFolder(rootFolder, 0);
+		
+		// create a new Gson instance
+		 Gson gson = new Gson();
+		 // convert BoxFolder list to json
+		 rv = gson.toJson(folders);
+		
+		return rv;
+	}
+	
+	/**
+	 * get list of BoxFolder object recursively
+	 * @param folder
+	 * @param depth
+	 * @return
+	 */
+	private static List<BoxFolder> listFolder(BoxFolder folder, int depth) {
+		List<BoxFolder> rv = new ArrayList<BoxFolder>();
+		
+		for (BoxItem.Info itemInfo : folder) {
+			String indent = "";
+			for (int i = 0; i < depth; i++) {
+				indent += "    ";
+			}
+
+			log.info(indent + itemInfo.getName());
+			if (itemInfo instanceof BoxFolder.Info) {
+				BoxFolder childFolder = (BoxFolder) itemInfo.getResource();
+				rv.add(childFolder);
+				if (depth < MAX_DEPTH) {
```

Comment:
Depend on how you define the depth...

With depth value starts with 0, and MAX_DEPTH set to one, this will return the root level folders and their first level child folders.